docs/Training-SAC.md

```diff
@@ -0,0 +1,348 @@
+# Training with Soft-Actor Critic
+
+In addition to [Proximal Policy Optimization (PPO)](Training-PPO.md), ML-Agents provides
+[Soft Actor-Critic](http://bair.berkeley.edu/blog/2018/12/14/sac/) to perform
+reinforcement learning.
+SAC uses several neural networks. During training, it uses two neural networks to learn
+the expected return of a given action when at a certain state. Then, a 3rd neural network
+learns the optimal policy that matches the probability of choosing an action to the
+Q function learned.
+
+In contrast with PPO, SAC is _off-policy_, i.e. can learn from experiences collected
+at any time during the past. As experiences are collected, they are placed in an
+experience replay buffer, and randomly drawn during training. This makes SAC
+significantly more sample-efficient, often requiring 5-10 times less samples to learn
+the same task as PPO. However, SAC tends to require more model updates. SAC is a
+good choice for heavier or slower environments (about 0.1 seconds per step or more).
+
+SAC is also a "maximum entropy" algorithm, and enables exploration in an intrinsic way.
+Read more about maximum entropy RL [here](https://bair.berkeley.edu/blog/2017/10/06/soft-q-learning/).
+
+To train an agent, you will need to provide the agent one or more reward signals which
+the agent should attempt to maximize. See [Reward Signals](Training-RewardSignals.md)
+for the available reward signals and the corresponding hyperparameters.
+
+See [Training ML-Agents](Training-ML-Agents.md) for instructions on running the
+training program, `learn.py`.
+
+If you are using the recurrent neural network (RNN) to utilize memory, see
+[Using Recurrent Neural Networks](Feature-Memory.md) for RNN-specific training
+details.
+
+If you are using curriculum training to pace the difficulty of the learning task
+presented to an agent, see [Training with Curriculum
+Learning](Training-Curriculum-Learning.md).
+
+For information about imitation learning from demonstrations, see
+[Training with Imitation Learning](Training-Imitation-Learning.md).
+
+## Best Practices when training with SAC
+
+Successfully training a Reinforcement Learning model often involves tuning the
+training hyperparameters. This guide contains some best practices for tuning the
+training process when the default parameters don't seem to be giving the level
```

Comment:
tuning the training process reads a bit odd.  "best practices for training when .."